### PR TITLE
Implement BLOCKHASH, TIMESTAMP, and BLOCKHEIGHT EVM opcodes

### PR DESCRIPTION
**Pull request by dplumb94**

### CODE REVIEW

File: families/seth/src/sawtooth_seth/common/constants.go [hunk 1]
```diff
@@ -29,6 +29,9 @@ const (
 	FAMILY_NAME    = "seth"
 	FAMILY_VERSION = "1.0"
 	ENCODING       = "application/protobuf"
+	BLOCK_INFO_PREFIX = "00b10c"
+    BLOCK_INFO_NAMESPACE = BLOCK_INFO_PREFIX + "00"
```
Comment (vaporos) [inline families/seth/src/sawtooth_seth/common/constants.go]: indent issue?

Comment (dplumb94) [reply families/seth/src/sawtooth_seth/common/constants.go]: I blame `gofmt`

File: families/seth/src/burrow/evm/vm.go [hunk 2]
```diff
@@ -640,20 +640,51 @@ func (vm *VM) call(caller, callee *Account, code, input []byte, value int64, gas
...
+				return nil, ErrInvalidContract
+			}
+			stack.Push64(time)
 			logger.Debugf(" => 0x%X\n", time)
 
 		case BLOCKHEIGHT: // 0x43
 			number := int64(vm.params.BlockHeight)
+			if number == 0 {
+				logger.Debugf(
+					" => TIMESTAMP instruction failed. Unable to get timestamp of requested block")
```
Comment (aludvik) [inline families/seth/src/burrow/evm/vm.go]: Update debug statement

File: families/seth/src/sawtooth_seth/common/state_address.go [hunk 3]
```diff
@@ -50,6 +50,16 @@ func NewStateAddrFromString(s string) (sa StateAddr, err error) {
 	return NewStateAddrFromBytes(bytes)
 }
 
+func NewBlockInfoAddr(n int64) (StateAddr, error) {
+	s := fmt.Sprintf("%x", n)
```
Comment (aludvik) [inline families/seth/src/sawtooth_seth/common/state_address.go]: Consider using `word256.Int64ToWord256`

File: families/seth/src/sawtooth_seth/processor/handler/state_manager.go [hunk 4]
```diff
@@ -165,3 +166,45 @@ func (mgr *StateManager) MustSetEntry(vmAddress *EvmAddr, entry *EvmEntry) {
...
+	entry := &BlockInfoConfig{}
+	err = proto.Unmarshal(entryData, entry)
+	if err != nil {
+		return nil, err
+	}
+
+	return entry, nil
+}
+
+func (mgr *StateManager) GetBlockInfo(blockInfoAddr string) (*BlockInfo, error) {
```
Comment (aludvik) [inline families/seth/src/sawtooth_seth/processor/handler/state_manager.go]: This should probably just take a uint64 as an argument and construct the address.

File: families/seth/src/burrow/word256/word.go [hunk 5]
```diff
@@ -50,6 +50,11 @@ func (w Word256) Compare(other Word256) int {
 	return bytes.Compare(w[:], other[:])
 }
 
+func StringToWord256(s string) Word256 {
+	bytes := []byte(s)
```
Comment (aludvik) [inline families/seth/src/burrow/word256/word.go]: I think this probably has to be `hex.DecodeString(s)`.

File: families/seth/src/sawtooth_seth/common/state_address.go (hunk 3 above)
Comment (dplumb94) [reply families/seth/src/sawtooth_seth/common/state_address.go]: Fixed as per our discussion

File: families/seth/src/burrow/evm/vm.go [hunk 6]
```diff
@@ -640,12 +640,33 @@ func (vm *VM) call(caller, callee *Account, code, input []byte, value int64, gas
...
+				if err != nil {
+					return nil, firstErr(err, ErrInvalidContract)
+				}
+				stack.Push(hash)
+			}
 
 		case COINBASE: // 0x41
 			stack.Push(Zero256)
-			logger.Debugf(" => 0x%X (NOT SUPPORTED)\n", stack.Peek().Bytes())
+			logger.Debugf(" => 0x%X \n", stack.Peek().Bytes())
```
Comment (aludvik) [inline families/seth/src/burrow/evm/vm.go]: Was this on purpose?

Comment (peterschwarz) [reply families/seth/src/burrow/evm/vm.go]: Probably should leave the `(NOT SUPPORTED)` in there

**Pull request merged**